src/engine/Algebra.ts

```diff
@@ -1,10 +1,13 @@
+/// <reference path="Interfaces/ConstructorBags/AlgebraOptions.ts" />
+
 module ex {
 
    /**
     * A 2D vector on a plane. 
     */
-   export class Vector {
-
+   export class Vector implements IVectorOptions {
```

Comment:
@kamranayub I didn't mean changing the constructor to take a `Vector`, I agree that should be `IVectorOptions`.

I was referring to having `implements IVectorOptions` on the `Vector` class.  With the constructor taking `IVectorOptions`, you can already in any object that satisfies the interface without having to explicitly implement `IVectorOptions`.